### ISSUE TITLE
Special characters in root password causing failure for metrics exporter

### ISSUE THREAD

**Issue by dfry** [bitnami/charts#22066]
### Name and Version

bitnami/mongodb 13.3.1

### What architecture are you using?

amd64

### What steps will reproduce the bug?

we create random passwords that include special characters for the mongodb chart install
For this particular random password: `4iw62X%*heiI%16Zxi2u`
The mongodb container runs fine, however, the metrics exporter fails giving the following error:
```
/opt/hostedtoolcache/go/1.17.12/x64/src/net/http/server.go:3034 +0x4e8
time="2024-01-12T14:12:09Z" level=error msg="Cannot connect to MongoDB: invalid MongoDB options: error parsing uri: invalid password: invalid URL escape \"%*h\""
```
Another example password that fails: `aTd7d7xUo88vNNS%x#tb` fails with this message
```
time="2024-01-12T14:12:06Z" level=error msg="Cannot connect to MongoDB: invalid MongoDB options: error parsing uri: invalid password: invalid URL escape \"%x#\""
```
### Are you using any custom parameters or values?

```yaml
global:
  storageClass: longhorn
nameOverride: cep-mongodb

fullnameOverride: ""

architecture: standalone
useStatefulSet: false

auth:
  enabled: true
  rootUser: root
  usernames: []
  passwords: []
  databases: []
  database: mojaloop
  username: mojaloop
  replicaSetKey: ""
  existingSecret: "mongodb-cep-secret"

persistence:
  enabled: true
  size: 8Gi
service:
  ports:
    mongodb: 27017
volumePermissions:
  enabled: true

metrics:
  enabled: true
  serviceMonitor:
    enabled: true
```

### What is the expected behavior?

the metrics exporter container should run without error

### What do you see instead?

as mentioned, exporter container crashes

**Comment by fmulero** [comment]
Hi @dfry

I've just reproduced your issue. The problem seems to be here: https://github.com/bitnami/charts/blob/947458bc7c22d489a506306ba48782d38b906450/bitnami/mongodb/templates/_helpers.tpl#L424-L442

As you can see in that code we are taking care of some particular characters but we can not apply a proper URL encode to cover all cases. Please feel free to open a PR covering your needs, we will be happy to review it.

As workaround you can set directly the `metrics.args` in your values, but that option is not very safe.

**Comment by dfry** [comment]
@fmulero any feedback on my last comment?

**Comment by fmulero** [comment]
Hi @dfry 

I ran some tests changing the helper I pointed out in my [previous comment](https://github.com/bitnami/charts/issues/22066#issuecomment-1893307227)  but they are been interpreted, even if we scaped them, as part of the login URL. If that the case, we can not do anything from our side, the problem is in the percona client. Did you have the opportunity to test it setting directly the `metrics.args` value or changing the code I share in my previous comment?
